site/content/docs/main/build-from-source.md

```diff
@@ -98,6 +98,69 @@ make container
 ```
 _Note: To build build container images for both `velero` and `velero-restic-restore-helper`, run: `make all-containers`_
 
+### Publishing container images to a registry
+
+To publish container images to a registry, the following one time setup is necessary:
+
+1. If you are building cross platform container images
+    ```bash
+    $ docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+    ```
+1. Create and bootstrap a new docker buildx builder
+    ```bash
+    $ docker buildx create --use --name builder
+      builder
+    $ docker buildx inspect --bootstrap
+      [+] Building 2.6s (1/1) FINISHED
+      => [internal] booting buildkit                                2.6s
+      => => pulling image moby/buildkit:buildx-stable-1             1.9s
+      => => creating container buildx_buildkit_builder0             0.7s
+    Name:   builder
+    Driver: docker-container
+
+    Nodes:
+    Name:      builder0
+    Endpoint:  unix:///var/run/docker.sock
+    Status:    running
+    Platforms: linux/amd64, linux/arm64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
+    ```
+    NOTE:
+
+    Without the above setup, the output of `docker buildx inspect --bootstrap` will be
+    ```bash
+    $ docker buildx inspect --bootstrap
+    Name:   default
+    Driver: docker
+
+    Nodes:
+    Name:      default
+    Endpoint:  default
+    Status:    running
+    Platforms: linux/amd64, linux/arm64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
+    ```
+    And the `REGISTRY=myrepo BUILDX_OUTPUT_TYPE=registry make container` will fail withthe below error
+    ```bash
+    $ REGISTRY=ashishamarnath BUILDX_PLATFORMS=linux/arm64 BUILDX_OUTPUT_TYPE=registry make container
+    auto-push is currently not implemented for docker driver
+    make: *** [container] Error 1
+    ```
+
+Having completed the above one time setup, now the output of `docker buildx inspect --bootstrap` should be like
```

Comment:
I think there should be a newline between the prose and the code blocks throughout, unless the new Markdown renderer is more generous with spacing.

Reply:
the preview looked good. But I am going to add the new lines. Better for human consumption.